topology/topology_parameter.h

```diff
@@ -456,6 +456,66 @@ class Gaussian2DParameter : public TopologyParameter
   double c_, p_center_, mean_x_, sigma_x_, mean_y_, sigma_y_, rho_;
 };
 
+/**
+ * Gamma parameter p(d) = d^(kappa-1)*exp(-d/theta)/(theta^kappa*Gamma(kappa))
+ */
+class GammaParameter : public RadialParameter
+{
+public:
+  /**
+   * Parameters:
+   * kappa    - shape
+   * theta    - scale
+   */
+  GammaParameter( const DictionaryDatum& d )
+    : RadialParameter( d )
+    , kappa_( 1.0 )
+    , theta_( 1.0 )
+  {
+    updateValue< double >( d, names::kappa, kappa_ );
+    updateValue< double >( d, names::theta, theta_ );
+    if ( kappa_ < 1 )
+      throw BadProperty(
+        "topology::GammaParameter: "
+        "kappa >= 1 required." );
+    if ( theta_ < 1 )
+      throw BadProperty(
+        "topology::GammaParameter: "
+        "theta >= 1 required." );
+
+    beta_ = 1. / theta_;
+    delta_ = std::pow( beta_, kappa_ ) / factorial( kappa_ - 1 );
+  }
+  double beta_;
+  double delta_;
+
+  int
+  factorial( int n )
+  {
+    int i, x = 1;
+    for ( i = 1; i <= n; i++ )
+    {
+      x *= i;
+    }
+    return x;
+  }
```

Comment:
This implementation of the factorial only works for integers. However, `kappa_` can be any `double` value. Passing it to the `factorial` function will then implicitly convert it to an integer, and if `kappa_` is a decimal number, we get the wrong value for `delta_`. I suggest instead using the [Spouge's approximation](https://en.wikipedia.org/wiki/Spouge%27s_approximation). You can find an example implementation [here](https://rosettacode.org/wiki/Gamma_function#C).